scripts/reformat-shfmt

```diff
@@ -15,6 +15,6 @@ if ! which shfmt > /dev/null; then
 	exit 1
 fi
 
-for file in $(shfmt -f scripts tests/shell); do
-	shfmt -s -sr -w "$file"
-done
+shell_files=$(shfmt -f 'scripts' 'tests/shell')
+non_ignored=$(printf "%s\n" "$shell_files" | sed -nE 's/.*/"\0"/p' | xargs git check-ignore -vn | sed -ne 's/^::[ \t]*\(.*\)$/\1/p')
```

Comment:
`git check-ignore` has different semantics than `git ls-files`. Files which are not part of the git repo, but also not ignored, will yield true in `check-ignore` but not in `ls-files`.

Reply:
I can't use `git ls-files` with `shfmt -f`. For some reason if you call `shfmt -f` with a filename (not a directory) it always prints that filename even if the file isn't a shell file. For example:
```sh
shfmt -f README.md
#> README.md
```